File: Programmers/42578_위장.js
function solution(clothes) {
    var closet = clothes.reduce((a, c) => {
        a[c[1]] = (a[c[1]]) ? a[c[1]] + 1 : 1;
        return a;
    }, {});

    var item = Object.values(closet);
    if (item.length === 1) {
        return item[0];
    }

    var result = 1;
    item.forEach((i) => result *= (i + 1));

    return result - 1;
}

File: Programmers/42578_위장.java
import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {

        HashMap<String, Integer> hm = new HashMap<>();
        for (String[] strings : clothes) {
            hm.put(strings[1], hm.getOrDefault(strings[1], 0) + 1);
        }

        int result = 1;
        for(Integer i : hm.values()) {
            result *= (i + 1);
        }

        return result - 1;
    }
}